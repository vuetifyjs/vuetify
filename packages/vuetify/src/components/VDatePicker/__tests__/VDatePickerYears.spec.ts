// @ts-nocheck
/* eslint-disable */

// import VDatePickerYears from '../VDatePickerYears'
import {
  mount,
  MountOptions,
  Wrapper,
} from '@vue/test-utils'

describe.skip('VDatePickerYears.ts', () => {
  type Instance = InstanceType<typeof VDatePickerYears>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VDatePickerYears, {
        ...options,
        mocks: {
          $vuetify: {
            rtl: false,
            lang: {
              t: () => {},
            },
          },
        },
      })
    }
  })

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2000',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respect min/max props', async () => {
    const wrapper = mountFunction({
      propsData: {
        min: 1234,
        max: 1238,
      },
    })

    expect(wrapper.findAll('li:first-child').at(0).element.textContent).toBe('1238')
    expect(wrapper.findAll('li:last-child').at(0).element.textContent).toBe('1234')
  })

  it('should not allow min to be greater then max', async () => {
    const wrapper = mountFunction({
      propsData: {
        min: 1238,
        max: 1234,
      },
    })
    expect(wrapper.findAll('li')).toHaveLength(1)
    expect(wrapper.findAll('li').at(0).element.textContent).toBe('1234')
    expect(wrapper.findAll('li').at(0).element.textContent).toBe('1234')
  })

  it('should emit event on year click', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: 1999,
      },
    })

    const input = jest.fn()
    wrapper.vm.$on('input', input)

    wrapper.findAll('li.active + li').at(0).trigger('click')
    expect(input).toHaveBeenCalledWith(1998)
  })

  it('should format years', async () => {
    const wrapper = mountFunction({
      propsData: {
        format: year => `(${year})`,
        min: 1001,
        max: 1001,
      },
    })

    expect(wrapper.findAll('li').at(0).element.textContent).toBe('(1001)')
  })
})
