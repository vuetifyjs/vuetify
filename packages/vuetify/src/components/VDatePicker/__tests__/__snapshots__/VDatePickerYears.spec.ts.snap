// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDatePickerYears.ts should render component and match snapshot 1`] = `
<ul class="v-date-picker-years">
  <li class>
    2100
  </li>
  <li class>
    2099
  </li>
  <li class>
    2098
  </li>
  <li class>
    2097
  </li>
  <li class>
    2096
  </li>
  <li class>
    2095
  </li>
  <li class>
    2094
  </li>
  <li class>
    2093
  </li>
  <li class>
    2092
  </li>
  <li class>
    2091
  </li>
  <li class>
    2090
  </li>
  <li class>
    2089
  </li>
  <li class>
    2088
  </li>
  <li class>
    2087
  </li>
  <li class>
    2086
  </li>
  <li class>
    2085
  </li>
  <li class>
    2084
  </li>
  <li class>
    2083
  </li>
  <li class>
    2082
  </li>
  <li class>
    2081
  </li>
  <li class>
    2080
  </li>
  <li class>
    2079
  </li>
  <li class>
    2078
  </li>
  <li class>
    2077
  </li>
  <li class>
    2076
  </li>
  <li class>
    2075
  </li>
  <li class>
    2074
  </li>
  <li class>
    2073
  </li>
  <li class>
    2072
  </li>
  <li class>
    2071
  </li>
  <li class>
    2070
  </li>
  <li class>
    2069
  </li>
  <li class>
    2068
  </li>
  <li class>
    2067
  </li>
  <li class>
    2066
  </li>
  <li class>
    2065
  </li>
  <li class>
    2064
  </li>
  <li class>
    2063
  </li>
  <li class>
    2062
  </li>
  <li class>
    2061
  </li>
  <li class>
    2060
  </li>
  <li class>
    2059
  </li>
  <li class>
    2058
  </li>
  <li class>
    2057
  </li>
  <li class>
    2056
  </li>
  <li class>
    2055
  </li>
  <li class>
    2054
  </li>
  <li class>
    2053
  </li>
  <li class>
    2052
  </li>
  <li class>
    2051
  </li>
  <li class>
    2050
  </li>
  <li class>
    2049
  </li>
  <li class>
    2048
  </li>
  <li class>
    2047
  </li>
  <li class>
    2046
  </li>
  <li class>
    2045
  </li>
  <li class>
    2044
  </li>
  <li class>
    2043
  </li>
  <li class>
    2042
  </li>
  <li class>
    2041
  </li>
  <li class>
    2040
  </li>
  <li class>
    2039
  </li>
  <li class>
    2038
  </li>
  <li class>
    2037
  </li>
  <li class>
    2036
  </li>
  <li class>
    2035
  </li>
  <li class>
    2034
  </li>
  <li class>
    2033
  </li>
  <li class>
    2032
  </li>
  <li class>
    2031
  </li>
  <li class>
    2030
  </li>
  <li class>
    2029
  </li>
  <li class>
    2028
  </li>
  <li class>
    2027
  </li>
  <li class>
    2026
  </li>
  <li class>
    2025
  </li>
  <li class>
    2024
  </li>
  <li class>
    2023
  </li>
  <li class>
    2022
  </li>
  <li class>
    2021
  </li>
  <li class>
    2020
  </li>
  <li class>
    2019
  </li>
  <li class>
    2018
  </li>
  <li class>
    2017
  </li>
  <li class>
    2016
  </li>
  <li class>
    2015
  </li>
  <li class>
    2014
  </li>
  <li class>
    2013
  </li>
  <li class>
    2012
  </li>
  <li class>
    2011
  </li>
  <li class>
    2010
  </li>
  <li class>
    2009
  </li>
  <li class>
    2008
  </li>
  <li class>
    2007
  </li>
  <li class>
    2006
  </li>
  <li class>
    2005
  </li>
  <li class>
    2004
  </li>
  <li class>
    2003
  </li>
  <li class>
    2002
  </li>
  <li class>
    2001
  </li>
  <li class="active primary--text">
    2000
  </li>
  <li class>
    1999
  </li>
  <li class>
    1998
  </li>
  <li class>
    1997
  </li>
  <li class>
    1996
  </li>
  <li class>
    1995
  </li>
  <li class>
    1994
  </li>
  <li class>
    1993
  </li>
  <li class>
    1992
  </li>
  <li class>
    1991
  </li>
  <li class>
    1990
  </li>
  <li class>
    1989
  </li>
  <li class>
    1988
  </li>
  <li class>
    1987
  </li>
  <li class>
    1986
  </li>
  <li class>
    1985
  </li>
  <li class>
    1984
  </li>
  <li class>
    1983
  </li>
  <li class>
    1982
  </li>
  <li class>
    1981
  </li>
  <li class>
    1980
  </li>
  <li class>
    1979
  </li>
  <li class>
    1978
  </li>
  <li class>
    1977
  </li>
  <li class>
    1976
  </li>
  <li class>
    1975
  </li>
  <li class>
    1974
  </li>
  <li class>
    1973
  </li>
  <li class>
    1972
  </li>
  <li class>
    1971
  </li>
  <li class>
    1970
  </li>
  <li class>
    1969
  </li>
  <li class>
    1968
  </li>
  <li class>
    1967
  </li>
  <li class>
    1966
  </li>
  <li class>
    1965
  </li>
  <li class>
    1964
  </li>
  <li class>
    1963
  </li>
  <li class>
    1962
  </li>
  <li class>
    1961
  </li>
  <li class>
    1960
  </li>
  <li class>
    1959
  </li>
  <li class>
    1958
  </li>
  <li class>
    1957
  </li>
  <li class>
    1956
  </li>
  <li class>
    1955
  </li>
  <li class>
    1954
  </li>
  <li class>
    1953
  </li>
  <li class>
    1952
  </li>
  <li class>
    1951
  </li>
  <li class>
    1950
  </li>
  <li class>
    1949
  </li>
  <li class>
    1948
  </li>
  <li class>
    1947
  </li>
  <li class>
    1946
  </li>
  <li class>
    1945
  </li>
  <li class>
    1944
  </li>
  <li class>
    1943
  </li>
  <li class>
    1942
  </li>
  <li class>
    1941
  </li>
  <li class>
    1940
  </li>
  <li class>
    1939
  </li>
  <li class>
    1938
  </li>
  <li class>
    1937
  </li>
  <li class>
    1936
  </li>
  <li class>
    1935
  </li>
  <li class>
    1934
  </li>
  <li class>
    1933
  </li>
  <li class>
    1932
  </li>
  <li class>
    1931
  </li>
  <li class>
    1930
  </li>
  <li class>
    1929
  </li>
  <li class>
    1928
  </li>
  <li class>
    1927
  </li>
  <li class>
    1926
  </li>
  <li class>
    1925
  </li>
  <li class>
    1924
  </li>
  <li class>
    1923
  </li>
  <li class>
    1922
  </li>
  <li class>
    1921
  </li>
  <li class>
    1920
  </li>
  <li class>
    1919
  </li>
  <li class>
    1918
  </li>
  <li class>
    1917
  </li>
  <li class>
    1916
  </li>
  <li class>
    1915
  </li>
  <li class>
    1914
  </li>
  <li class>
    1913
  </li>
  <li class>
    1912
  </li>
  <li class>
    1911
  </li>
  <li class>
    1910
  </li>
  <li class>
    1909
  </li>
  <li class>
    1908
  </li>
  <li class>
    1907
  </li>
  <li class>
    1906
  </li>
  <li class>
    1905
  </li>
  <li class>
    1904
  </li>
  <li class>
    1903
  </li>
  <li class>
    1902
  </li>
  <li class>
    1901
  </li>
  <li class>
    1900
  </li>
</ul>
`;
